clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -229,7 +229,7 @@ public long connectionDelay(Node node, long now) {
      */
     @Override
     public boolean connectionFailed(Node node) {
-        return connectionStates.connectionState(node.idString()).equals(ConnectionState.DISCONNECTED);
+        return connectionStates.isDisconnected(node.idString());
```

Comment:
@ijuma I'm not sure, but I think that might not be sufficient. The problem is that we'd only see disconnects for connections with in-flight requests. However, we need to see all disconnects in order to be able to clear out the unsent requests. 

Maybe we just need to expose the disconnected nodes directly from `NetworkClient`? 
```
List<Node> disconnected();
```